conan.cmake

```
@@ -296,7 +296,18 @@ function(conan_cmake_detect_unix_libcxx result)
 endfunction()
 
 function(conan_cmake_detect_vs_runtime result)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
+    if(ARGUMENTS_BUILD_TYPE)
+        if (${ARGUMENTS_BUILD_TYPE} STREQUAL "None")
+            set(build_type ${CMAKE_BUILD_TYPE})
+        else()
+            set(build_type ${ARGUMENTS_BUILD_TYPE})
+        endif()    
+    elseif(CMAKE_BUILD_TYPE)
+        set(build_type ${CMAKE_BUILD_TYPE})
+    endif()
+    if(build_type)
+        string(TOUPPER build_type build_type)
```

Sorry to jump in but this is a small typo I think 😄 
I used this branch to fix some issues I was having using libcurl in MSVC debug mode, and noticed this.
```suggestion
        string(TOUPPER ${build_type} build_type)
```

---

Hi @CrustyAuklet , yes, it was a typo, thanks a lot for pointing this out!